.github/workflows/pullrequest.yml

```diff
@@ -20,15 +21,16 @@ jobs:
           files_ignore: '.github/**'
       - name: List all changed files
         run: |
-          for FILE in ${{ steps.changed-files.outputs.all_changed_files }}; do
-            REGEX=".*\.(.*)"
-            if [[  "$FILE" =~ $REGEX ]]
-            then
-                EXTENSION=${BASH_REMATCH[1]}
-                EXTENSIONS_LIST=($EXTENSIONS)
-                if [[ " ${EXTENSIONS_LIST[*]} " =~ " ${EXTENSION} " ]]
-                then
-                    .github/verify_header.sh $FILE
-                fi
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            for file_in_list in $FILES; do
```

Comment:
I think you can declare $FILES as an array rather than a string:
```
FILES=(*Makefile)
```
... and then explicitly check whether element is on the list instead of the inner loop:
```
if [[ " ${FILES[*]} " =~ " $file " ]]; then
```

Reply:
Yes I though about it also, but the reason I decided to make it like I did is how regular expression actually work here. Or more precisely - what is considered to be a wildcard here.
To match any Makefile we would need to specify it as `Makefile$` instead of `*Makefile`. While it is a small change here, it will be much more complicated in the same PR in OCL (https://github.com/Open-CAS/open-cas-linux/pull/1203/files) where we have more files included in this variable. For example `configure` should be put as `^configure$` to match only this file and not any file that has a 'configure' string in his full path name.
This could be done of course, but it will be much more complicated for future modifications while adding more files into it. I think this way is more user friendly. And this additional loop is not much of a performance overhead also, because we are testing there only files we put into FILES variable against files we modified in this PR (and not entire repo).